config/cmake_ext_mod/HDFMacros.cmake

```diff
@@ -29,9 +29,9 @@ macro (SET_HDF_BUILD_TYPE)
   endif()
   if(NOT CMAKE_BUILD_TYPE AND NOT CMAKE_CONFIGURATION_TYPES)
     if (CMAKE_VERSION VERSION_GREATER_EQUAL "3.15.0")
-      message (VERBOSE "Setting build type to 'RelWithDebInfo' as none was specified.")
+      message (VERBOSE "Setting build type to 'Debug' as none was specified.")
     endif()
-    set(CMAKE_BUILD_TYPE RelWithDebInfo CACHE STRING "Choose the type of build." FORCE)
+    set(CMAKE_BUILD_TYPE Debug CACHE STRING "Choose the type of build." FORCE)
     # Set the possible values of build type for cmake-gui
```

Comment:
This should be Debug by default for develop, not RelWithDebInfo, for parity with the Autotools.